modules/Link.js

```diff
@@ -20,6 +20,21 @@ function isEmptyObject(object) {
   return true
 }
 
+function isLinkActive(router, props) {
```

Comment:
In my testing this doesn’t seem to be the case. I think the listener fires before React has a chance to process `setState` in the router so it looks like everything happens in a single batch. I might be wrong about the exact mechanics though—just saying I don’t see duplicate `render`s in my example app.